Tasks/Gradle/CodeAnalysis/Common/CodeAnalysisResultPublisher.ts

```diff
@@ -0,0 +1,144 @@
+import {AnalysisResult} from './AnalysisResult'
+
+import path = require('path');
+import fs = require('fs');
+import glob = require('glob');
+
+import tl = require('vsts-task-lib/task');
+
+
+export class CodeAnalysisResultPublisher {
+    constructor(private analysisResults: AnalysisResult[], private stagingDir: string) {
+        if (!analysisResults) {
+            throw new ReferenceError('analysisResults');
+        }
+
+        if (!stagingDir) {
+            throw new ReferenceError('stagingDir');
+        }
+    }
+
+    /**
+     * Uploads the artifacts. It groups them by module
+     * 
+     * @param {string} prefix
```

Comment:
It's used to discriminate between artifacts comming from different builds of the same projects. So an artifact name will be 1234_Module1_PMD.xml   -- helps with downloading the file.